src/app/ReadHandler.cpp

```diff
@@ -31,13 +31,28 @@
 #include <lib/core/TLVUtilities.h>
 #include <messaging/ExchangeContext.h>
 
+#include <app-common/zap-generated/attributes/Accessors.h>
 #include <app/ReadHandler.h>
 #include <app/reporting/Engine.h>
 
 namespace chip {
 namespace app {
 using Status = Protocols::InteractionModel::Status;
 
+CHIP_ERROR ReadHandler::GetPublisherSelectedIntervalLimit(uint16_t & limit_s)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_SED
+    uint32_t idle_interval_ms = 0;
+    EmberAfStatus status =
+        chip::app::Clusters::IcdManagement::Attributes::IdleModeInterval::Get(kRootEndpointId, &idle_interval_ms);
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == status, CHIP_ERROR_INTERNAL);
```

Comment:
This is layer-crossing and the wrong method. The source of information for the attribute store should be the same as what is read here, but the IM read handler should not try to read the attribute store. The attribute store is an implementation detail of clusters, not a fundamental data structure to be used in other layers. Please find a way to inject this in the ReadHandler init, and have the IdleModeInterval value be reported by the attribute from the same source of truth.

Reply:
Another way of saying it: the getters/accessors like `chip::app::Clusters::IcdManagement::Attributes::IdleModeInterval::Get` have two major major issues:

- They are process global, so cannot be used if using multiple instances of Matter node servers in 1 process (this is very hard to refactor, the more of this there are)
- They only work for Attribute-store backed attributes, not for AttributeAccessInterfcace-backed attributes. The generalization of the internal API to read *any* value of the data model, even code backed, would rely on locally calling into IM API (which trickles down to read handler eventually) to access a particular attribute, letting the path to the attribute reach the attribute store or code-backed cluster at the next level down.

By reaching into attribute store here, there is a circular dependency that would forever prevent (without refactor) the ability to read arbitrary server attributes (including IdleModeInterval) via a single API surface.

Reply:
I understand your concern, in this case, this value is Attribute-backed. The attribute is fixed, so the source of truth is the attribute value itself, set by the configuration.
Do we even have such IM API for handling attributes?